File: fib.go
package main

import (
    "fmt"
    "math/big"
    "runtime"
    "time"
)

func fibonacci(n int) *big.Int {
    if n <= 1 {
        return big.NewInt(int64(n))
    }
    a := big.NewInt(0)
    b := big.NewInt(1)
    temp := big.NewInt(0)
    for i := 2; i <= n; i++ {
        temp.Add(a, b)
        a.Set(b)
        b.Set(temp)
    }
    return b
}

func main() {
    var m runtime.MemStats
    runtime.ReadMemStats(&m)

    start := time.Now()
    result := fibonacci(100)
    elapsed := time.Since(start)

    runtime.ReadMemStats(&m)
    memoryKB := m.Alloc / 1024

    fmt.Println("Language: Go")
    fmt.Printf("Time: %.6f seconds\n", elapsed.Seconds())
    fmt.Printf("Memory: %d KB\n", memoryKB)
    fmt.Printf("Result: %s\n", result.String())
}

File: fib.dart
import 'dart:io';

int fibonacci(int n) {
  if (n <= 1) return n;
  int a = 0, b = 1, temp;
  for (int i = 2; i <= n; i++) {
    temp = a + b;
    a = b;
    b = temp;
  }
  return b;
}

void main() {
  final stopwatch = Stopwatch()..start();
  final result = fibonacci(50);
  stopwatch.stop();
  
  final memoryInfo = ProcessInfo.currentRss;
  final memoryKB = memoryInfo ~/ 1024;
  
  print('Language: Dart');
  print('Time: ${(stopwatch.elapsedMicroseconds / 1000000).toStringAsFixed(6)} seconds');
  print('Memory: $memoryKB KB');
  print('Result: $result');
}
